flake.nix

```
@@ -38,14 +38,35 @@
               });
 
             src = ./.;
-          });
+
+            strictDeps = true;
+            nativeBuildInputs = [ which ];
+            buildInputs = [ zlib xz zstd lz4 lzo ];
+
+            preBuild = ''
+              cd squashfs-tools
+            '';
+
+            installFlags = [
+              "INSTALL_DIR=${placeholder "out"}/bin"
+              "INSTALL_MANPAGES_DIR=${placeholder "out"}/share/man/man1"
+            ];
+
+            makeFlags = [
+              "XZ_SUPPORT=1"
+              "ZSTD_SUPPORT=1"
+              "LZ4_SUPPORT=1"
+              "LZMA_SUPPORT=1"
+              "LZO_SUPPORT=1"
+            ];
```

I would add GZIP to be fully explicity about what compression we support:

```suggestion
            makeFlags = [
              "GZIP_SUPPORT=1"
              "XZ_SUPPORT=1"
              "ZSTD_SUPPORT=1"
              "LZ4_SUPPORT=1"
              "LZMA_SUPPORT=1"
              "LZO_SUPPORT=1"
            ];
```